### DISCUSSION TITLE
Log4Net: Application is unusable when Seq server is unavailable

### DISCUSSION THREAD

**Discussion by prism-defence** [datalust/seq-tickets#2056] [Help]
We are using seq with log4net.async and write testing software that is used in the field, so we don't always have access to good networks. We have found that if the seq server is unavailable, the performance of our software degrades to the point that its unusable (we know it's seq, as when we comment out the appender everything is fine again). Is there any way to improve the performance in this circumstance, e.g. get seq to fail if the server is unavailable, rather than then keep trying to connect as it seems to be doing? We have found that increasing the buffer size improves things, but isn't that just delaying the issue until the buffer gets full?

**Comment by nblumhardt** [comment]
Hi! Are you using Log4Net.Async's `ParallelForwardingAppender`, or the `AsyncForwardingAppender`? If it's the "parallel" one, you might be hitting:

> If the queue fills as a result of the rate of message creation exceeding the rate of appending, threads creating messages will block

(From https://github.com/cjbhaines/Log4Net.Async)

Log4Net.Async has been deprecated by its author; do your circumstances support a move to a newer logging framework? NLog and Serilog are both comparable in functionality and are much more actively maintained/better supported.

Happy to dig in otherwise :-)

**Reply by prism-defence** [reply]
We use AsyncForwardingAppender. log4net is used throughout our code base, so I doubt there'll be any enthusiasm to change, so if it is possible to find a solution that would be very much appreciated.

**Reply by nblumhardt** [reply]
Thanks for your reply! I can take a closer look :+1: 

Can you please mail me via `support@datalust.co` and include:

 - your target framework and operating system
 - app model, e.g. WinForms/WPF/embedded/...
 - log4net, log4net.async, seq.client.log4net package versions
 - log4net XML configuration and bootstrapping code from the app
 - example log event/logging statement from the app

This should be enough for me to set up a local repro, but anything else you can think to include would help.

Please also send your Seq subscription id if you have one (we're happy providing best-effort support for the Individual version, but can allocate more time for subscribers).
